### DISCUSSION TITLE
Can we initialise AzurePushHub on some page rather than in MauiProgram ?

### DISCUSSION THREAD

**Discussion by MGohil** [shinyorg/shiny#1288] [Q&A] [answered]
Hi,

I came across a situation where I would like to initialize the AzurePushHub on some page after login and not on MauiProgram builder.
This sample shows it.
https://github.com/shinyorg/shiny/blob/master/samples/Sample.Push.Maui/MauiProgram.cs#L46

Example, following is the code from MauiProgram, where I am initialising the hub, but I want to move this to some Page (HomePage)
```
var pushHubConfig = builder.Configuration
            .GetSection("AzurePushNotificationHub");

builder.Services.AddPushAzureNotificationHubs<PushNotificationDelegate>(
            pushHubConfig["ListenerConnectionString"],
            pushHubConfig["HubName"]);
```

The reason I would like to move this to page is, I am to read the Connection String and HubName from somewhere (Ex. Database or API or SecureStorage). But I can not get these in MauiProgram.

Kindly help to know if there is a way.

Thanks,
Milan G

**Comment by aritchie** [comment, marked as answer]
No - you currently cannot do this

**Reply by aritchie** [reply]
It is now documented how to create your own push provider for additional logic: https://shinylib.net/client/push/providers/

To those posting confused on my post, I don't care if you don't like the answer that I don't have a particular feature - simple: love it or leave it.  I have absolute zero tolerance for any bs here!
